components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/MeApiServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package org.wso2.carbon.identity.rest.api.user.association.v1.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.rest.api.user.association.v1.MeApiService;
+import org.wso2.carbon.identity.rest.api.user.association.v1.core.UserAssociationService;
+import org.wso2.carbon.identity.rest.api.user.association.v1.dto.AssociationSwitchRequestDTO;
+import org.wso2.carbon.identity.rest.api.user.association.v1.dto.AssociationUserRequestDTO;
+import org.wso2.carbon.identity.rest.api.user.association.v1.dto.UserDTO;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.rest.api.user.association.v1.AssociationEndpointConstants.ASSOCIATIONS_PATH;
+
+public class MeApiServiceImpl extends MeApiService {
+
+    @Autowired
+    private UserAssociationService userAssociationService;
+
+    @Override
+    public Response meAssociationsDelete() {
+
+        userAssociationService.deleteUserAccountAssociation(getUserId());
+        return Response.ok().build();
+    }
+
+    @Override
+    public Response meAssociationsGet() {
+
+        List<UserDTO> userDTOList = userAssociationService.getAssociationsOfUser(getUserId());
+        if (userDTOList.size() == 0) {
+            return Response.noContent().build();
+        }
+        return Response.ok().entity(userDTOList).build();
+    }
+
+    @Override
+    public Response meAssociationsPost(AssociationUserRequestDTO association) {
+
+        userAssociationService.createUserAccountAssociation(association);
+        try {
+            return Response.created(getAssociationsLocationURI(getUserId())).build();
+        } catch (URISyntaxException e) {
+            return Response.status(Response.Status.CREATED).build();
+        }
+    }
+
+    @Override
+    public Response meAssociationsSwitchPut(AssociationSwitchRequestDTO switchUserReqeust) {
+
+        boolean isSwitched = userAssociationService.switchLoggedInUser(getUserId());
+        if (isSwitched) {
+            return Response.ok().build();
+        } else {
+            //TODO need to check this.
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+    private String getUserId() {
+
+        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        return UserCoreUtil.addTenantDomainToEntry(username, PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getTenantDomain());
+    }
+
+    private URI getAssociationsLocationURI(String userId) throws URISyntaxException {
+
+        return new URI("/" + userId + "/" + ASSOCIATIONS_PATH);
```

Comment:
use the util in https://github.com/wso2/identity-api-user/blob/master/components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java#L109 to build fully qualified path

Reply:
done